archive/s/scala/factorial.scala

```diff
@@ -0,0 +1,23 @@
+// Scala Program to calculate Factorial 
+// of a number using recursion 
+
+// Creating object 
+object compile
+{ 
+	// Function to calculate 
+	// factorial using Recursive 
+	// formula (i.e N! = N * N-1 !) 
+	def factorial(n: Int): Int =
+	{ 
+		if (n == 0) 
+			return 1
+		else
+			return n * factorial(n-1) 
+	} 
+
+	// Driver Code 
+	def main(args: Array[String]) 
+	{ 
+		println(factorial(5)) 
```

Comment:
Can this pull from the command line (`args`) instead of hardcoding 5?

Reply:
okay I'll get back to in some time with the updated code.